packages/vite/src/index.ts

```diff
@@ -137,19 +137,24 @@ export function imagetools(userOptions: Partial<VitePluginOptions> = {}): Plugin
     },
 
     renderChunk(code) {
-      const assetUrlRE = /__VITE_IMAGE_ASSET__([a-z\d]{8})__(?:_(.*?)__)?/g
+      const assetUrlRE = /const\s+(\S+)\s*=\s*"((__VITE_ASSET__([a-z\d]{8})__ \d+w,?\s?)+)";/g
 
       let match
       let s
       while ((match = assetUrlRE.exec(code))) {
         s = s || (s = new MagicString(code))
-        const [full, hash, postfix = ''] = match
-
-        const file = this.getFileName(hash)
-
-        const outputFilepath = viteConfig.base + file + postfix
-
-        s.overwrite(match.index, match.index + full.length, outputFilepath)
+        const [full, name, assets] = match
```

Comment:
a code comment here would be nice. e.g. it'd be helpful to know what the constructed value is supposed to look like. i'm also curious if there's a section of Vite's code that this is mirroring that we could link to